demos/ASP.Net Core API Gateway/Divergent.Sales.ViewModelComposition/OrdersListViewModelAppender.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Divergent.Sales.ViewModelComposition.Events;
+using ITOps.ViewModelComposition;
+using ITOps.ViewModelComposition.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Divergent.Sales.ViewModelComposition
+{
+    public class OrdersListViewModelAppender : IViewModelAppender
+    {
+        public bool Matches(RouteData routeData, string verb)
+        {
+            /*
+             * matching is a bit weak in this sample, it's designed 
+             * this way to satisfy both the Gateway and the Mvc sample
+             */
+            var controller = (string)routeData.Values["controller"];
+
+            return HttpMethods.IsGet(verb)
+                && controller.ToLowerInvariant() == "orders"
+                && !routeData.Values.ContainsKey("id");
+        }
+
+        public async Task Append(dynamic vm, RouteData routeData, IQueryCollection query)
```

Comment:
First of all, I don't see that as a big issue but what the coordinator could do is:
a. simply append vm returned from appenders to the response (no knowledge about appenders needed)
b. detect clashes and do some namespacing (add a json-node derived from appender name etc.)